main.tf

```
@@ -361,3 +387,76 @@ resource "aws_s3_access_point" "ap" {
 
   depends_on = [var.module_depends_on]
 }
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Replication Role and policy
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_iam_role" "replication" {
+  count = local.replication_enabled ? 1 : 0
+  name  = "${var.bucket}-replication-role"
+
+  assume_role_policy = jsonencode(
+    {
+      "Version" : "2012-10-17",
+      "Statement" : [
+        {
+          "Action" : "sts:AssumeRole",
+          "Principal" : {
+            "Service" : "s3.amazonaws.com"
+          },
+          "Effect" : "Allow",
+          "Sid" : ""
+        }
+      ]
+    }
+  )
+}
+
+resource "aws_iam_policy" "replication" {
+  count = local.replication_enabled ? 1 : 0
+  name  = "${var.bucket}-replication-policy"
+
+  policy = jsonencode(
+    {
+      "Version" : "2012-10-17",
+      "Statement" : [
+        {
+          "Action" : [
+            "s3:GetReplicationConfiguration",
+            "s3:ListBucket",
+            "s3:GetObjectVersion",
+            "s3:GetObjectVersionAcl",
+            "s3:ReplicateObject",
+            "s3:ReplicateDelete"
+          ],
+          "Effect" : "Allow",
+          "Resource" : [
+            "${local.bucket_arn}",
+            "${local.bucket_arn}/*"
+          ]
+        },
+        {
+          "Action" : [
+            "s3:GetBucketVersioning",
+            "s3:PutBucketVersioning",
+            "s3:ReplicateObject",
+            "s3:ReplicateDelete",
+            "s3:ObjectOwnerOverrideToBucketOwner"
+          ],
+          "Effect" : "Allow",
+          "Resource" : [
+            "arn:aws:s3:::${var.replication.bucket}/*",
+            "arn:aws:s3:::${var.replication.bucket}/"
+          ]
+        }
```

according to the example in the provider this should be:

also https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-walkthrough1.html specifies the same..

where are your policies taken from?

```suggestion
        {
          "Action" : [
            "s3:ReplicateObject",
            "s3:ReplicateDelete",
            "s3:ReplicateTags",
          ],
          "Effect" : "Allow",
          "Resource" : [
            "arn:aws:s3:::${var.replication.bucket}/*",
          ]
        }
```